### DISCUSSION TITLE
Try to run new-relic

### DISCUSSION THREAD

**Discussion by experttool** [paketo-buildpacks/feedback#312] [Java Team]
I'm trying to build and run SpringBoot application + NewRelic agent (using paketo buildpacks)and my pod won't start.
The following error occurs when contrainer starts:

```
ERROR: failed to launch: exec.d: failed to execute exec.d file at path '/layers/paketo-buildpacks_new-relic/helper/exec.d/properties': fork/exec /layers/paketo-buildpacks_new-relic/helper/exec.d/properties: exec format error
```

Here is my configuration

1. gradle task - i build it using 'gradlew bootBuildImage'

```
tasks.bootBuildImage {
    imageName = "aaa/bbb:${version}"
    builder = "paketobuildpacks/builder-jammy-base"
    buildpacks = listOf("urn:cnb:builder:paketo-buildpacks/java", "docker.io/paketobuildpacks/new-relic")
    bindings = listOf("${project.rootDir}/bindings/new-relic:/platform/bindings/new-relic:ro")
    publish = true
    @OptIn(ExperimentalEncodingApi::class) val auth =
        String(Base64.Default.decode(project.property("docker.password").toString()))
    docker {
        publishRegistry {
            username = project.property("docker.username").toString()
            password = auth
        }
    }
}
```

2. My bindings keys are: type, app_name, license_key, environment

3. On kubernetes I mapped secrets to volume and I have 4 files in directory /platform/bindings/new-relic

```
$ pwd
/platform/bindings/new-relic
$ ls
app_name  environment  license_key  type
$ 
```

**Comment by dmikusa** [comment]
Where are you performing the build? It almost seems like it's using an arm64 binary on a amd64 system. I think that's the error you see when the architectures get mixed up. Are you building the image on Macbook M-series hardware by chance?

**Reply by experttool** [reply]
Yes, it's built on Mac M, but I run it on Kubernetes/linux

**Reply by dmikusa** [reply]
Ok, it is probably getting confused because we do support ARM64 builds. There's not presently an option in Spring Boot Build Tools to force the platform, however, that's coming.

For the time being you'd need to build on an amd64/x86 system (Github Actions works well for this) or use `pack` cli to build, because it presently supports specifying a platform, to ensure that your containers are consistently built for your target environment which is amd64/x86.

**Reply by piotrblasiak** [reply]
> Ok, it is probably getting confused because we do support ARM64 builds. There's not presently an option in Spring Boot Build Tools to force the platform, however, that's coming.
> 
> For the time being you'd need to build on an amd64/x86 system (Github Actions works well for this) or use `pack` cli to build, because it presently supports specifying a platform, to ensure that your containers are consistently built for your target environment which is amd64/x86.

Do you have a link to a github issue where this is addresed so that we can follow the progress? Also, we have been building amd64 images on our apple silicon for years now - it´s only when adding this buildpack that it becomes a problem.

**Reply by dmikusa** [reply]
> Do you have a link to a github issue where this is addresed so that we can follow the progress?

It's in Spring Boot 3.4, https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.4.0-M2-Release-Notes#image-building-improvements

> Also, we have been building amd64 images on our apple silicon for years now - it´s only when adding this buildpack that it becomes a problem.

This buildpack includes a binary into the application image, which runs before your app starts up (to apply configuration for New Relic). The binary that gets added would be architecture-specific. You can see that is specifically what is failing in the example output listed above. 

It's hard to say beyond that and with the info available, some things have been evolving with buildpacks and the tooling as well, to enable support for ARM64. We did have reports of occasional issues there.

If you have time, try an upgrade to Spring Boot 3.4 M3 and see if you're still seeing issues.
